packet/checksum_test.go

```diff
@@ -39,146 +38,134 @@ CalculatePseudoHdrIPv6UDPCksum
 */
 
 func TestCalculateIPv4Checksum(t *testing.T) {
-	pkt := packet.GetPacket()
-	packet.InitEmptyIPv4TCPPacket(pkt, payloadSize)
-	initIPv4Addrs(pkt)
+	pkt := GetPacket()
+	InitEmptyIPv4TCPPacket(pkt, payloadSize2)
+	initIPv4Addrs2(pkt)
 
 	want := wantIPv4
-	got := packet.CalculateIPv4Checksum(pkt.GetIPv4())
+	got := CalculateIPv4Checksum(pkt.GetIPv4())
 	if got != want {
 		t.Errorf("Incorrect result:\ngot: %x, \nwant: %x\n\n", got, want)
 	}
 }
 
 func TestCalculateIPv4TCPChecksum(t *testing.T) {
-	pkt := packet.GetPacket()
-	packet.InitEmptyIPv4TCPPacket(pkt, payloadSize)
-	initIPv4Addrs(pkt)
+	pkt := GetPacket()
+	InitEmptyIPv4TCPPacket(pkt, payloadSize2)
+	initIPv4Addrs2(pkt)
 	initPorts(pkt)
 	initData(pkt)
 
-	ipcksum := packet.CalculateIPv4Checksum(pkt.GetIPv4())
-	pkt.GetIPv4().HdrChecksum = packet.SwapBytesUint16(ipcksum)
+	ipcksum := CalculateIPv4Checksum(pkt.GetIPv4())
+	pkt.GetIPv4().HdrChecksum = SwapBytesUint16(ipcksum)
 
 	want := wantIPv4TCP
-	got := packet.CalculateIPv4TCPChecksum(pkt.GetIPv4(), pkt.GetTCPForIPv4(), pkt.Data)
+	got := CalculateIPv4TCPChecksum(pkt.GetIPv4(), pkt.GetTCPForIPv4(), pkt.Data)
 	if got != want {
 		t.Errorf("Incorrect result:\ngot: %x, \nwant: %x\n\n", got, want)
 	}
 }
 
 func TestCalculateIPv6TCPChecksum(t *testing.T) {
-	pkt := packet.GetPacket()
-	packet.InitEmptyIPv6TCPPacket(pkt, payloadSize)
-	initIPv6Addrs(pkt)
+	pkt := GetPacket()
+	InitEmptyIPv6TCPPacket(pkt, payloadSize2)
+	initIPv6Addrs2(pkt)
 	initPorts(pkt)
 	initData(pkt)
 
 	want := wantIPv6TCP
-	got := packet.CalculateIPv6TCPChecksum(pkt.GetIPv6(), pkt.GetTCPForIPv6(), pkt.Data)
+	got := CalculateIPv6TCPChecksum(pkt.GetIPv6(), pkt.GetTCPForIPv6(), pkt.Data)
 	if got != want {
 		t.Errorf("Incorrect result:\ngot: %x, \nwant: %x\n\n", got, want)
 	}
 }
 
 func TestCalculateIPv4UDPChecksum(t *testing.T) {
-	pkt := packet.GetPacket()
-	packet.InitEmptyIPv4UDPPacket(pkt, payloadSize)
-	initIPv4Addrs(pkt)
+	pkt := GetPacket()
+	InitEmptyIPv4UDPPacket(pkt, payloadSize2)
+	initIPv4Addrs2(pkt)
 	initPorts(pkt)
 	initData(pkt)
 
-	ipcksum := packet.CalculateIPv4Checksum(pkt.GetIPv4())
-	pkt.GetIPv4().HdrChecksum = packet.SwapBytesUint16(ipcksum)
+	ipcksum := CalculateIPv4Checksum(pkt.GetIPv4())
+	pkt.GetIPv4().HdrChecksum = SwapBytesUint16(ipcksum)
 
 	want := wantIPv4UDP
-	got := packet.CalculateIPv4UDPChecksum(pkt.GetIPv4(), pkt.GetUDPForIPv4(), pkt.Data)
+	got := CalculateIPv4UDPChecksum(pkt.GetIPv4(), pkt.GetUDPForIPv4(), pkt.Data)
 	if got != want {
 		t.Errorf("Incorrect result:\ngot: %x, \nwant: %x\n\n", got, want)
 	}
 }
 
 func TestCalculateIPv6UDPChecksum(t *testing.T) {
-	pkt := packet.GetPacket()
-	packet.InitEmptyIPv6UDPPacket(pkt, payloadSize)
-	initIPv6Addrs(pkt)
+	pkt := GetPacket()
+	InitEmptyIPv6UDPPacket(pkt, payloadSize2)
```

Comment:
awful name! maybe better add local or smth making sence

Reply:
I will add "l" instead of 2.